cincinnati/src/plugins/internal/edge_add_remove.rs

```diff
@@ -127,11 +139,66 @@ impl EdgeAddRemovePlugin {
                 },
             )?;
 
+        // Remove edges instructed by "previous.remove_regex"
+        let previous_remove_regex_key = format!("{}.{}", self.key_prefix, "previous.remove_regex");
         graph
-            .find_by_metadata_key(&format!("{}.{}", self.key_prefix, "next.remove"))
+            .find_by_metadata_key(&previous_remove_regex_key)
+            .into_iter()
+            .try_for_each(
+                |(to, to_version, from_regex_string): (ReleaseId, String, String)| -> Fallible<()> {
+                    if self.remove_consumed_metadata {
+                        graph
+                            .get_metadata_as_ref_mut(&to)
+                            .map(|metadata| metadata.remove(&previous_remove_regex_key))?;
+                    }
+
+                    if from_regex_string == ".*" {
+                        let parents: Vec<daggy::EdgeIndex> = graph
+                            .previous_releases(&to)
+                            .map(|(edge_index, _, _)| edge_index)
+                            .collect();
+
+                        trace!("removing parents for '{}': {:?}", to_version, parents);
+                        return graph.remove_edges_by_index(&parents);
+                    };
+
+                    let from_regex = regex::Regex::new(&from_regex_string)
```

Comment:
This should happen before other `trace!` calls so that remove_regexp code path would be more distinct